### PR TITLE
cpr: Validate out cpr when using stdlibcxx

### PR DESCRIPTION
**Pull request by RubenRBS**

### CODE REVIEW

**Comment by perseoGI** [comment]
Compilation logs when using libstdc++ compiling cpr/1.10.0

<details>

```
conan install --requires "cpr/1.10.0" -s compiler.libcxx=libstdc++ --build="cpr/*" --update --build=missing

======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=gnu17
compiler.libcxx=libstdc++
compiler.version=15
os=Macos

Profile build:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=gnu17
compiler.libcxx=libc++
compiler.version=15
os=Macos


======== Computing dependency graph ========
cpr/1.10.0: Checking remote: conancenter
libcurl/8.0.1: Checking remote: conancenter
zlib/1.3.1: Checking remote: conancenter
libtool/2.4.7: Checking remote: conancenter
automake/1.16.5: Checking remote: conancenter
autoconf/2.71: Checking remote: conancenter
m4/1.4.19: Checking remote: conancenter
gnu-config/cci.20210814: Checking remote: conancenter
pkgconf/2.1.0: Checking remote: conancenter
meson/1.2.2: Checking remote: conancenter
ninja/1.11.1: Checking remote: conancenter
Graph root
    cli
Requirements
    cpr/1.10.0#0ff0b00f5ea92fd3590d64c462477bd2 - Cache (conancenter)
    libcurl/8.0.1#6a17c16cc45f13b1ecbeca1f2dcd537e - Cache (conancenter)
    zlib/1.3.1#f52e03ae3d251dec704634230cd806a2 - Cache (conancenter)
Build requirements
    autoconf/2.71#f9307992909d7fb3df459340f1932809 - Cache (conancenter)
    automake/1.16.5#058bda3e21c36c9aa8425daf3c1faf50 - Cache (conancenter)
    gnu-config/cci.20210814#dc430d754f465e8c74463019672fb97b - Cache (conancenter)
    libtool/2.4.7#08316dad5c72c541ed21e039e4cf217b - Cache (conancenter)
    m4/1.4.19#b38ced39a01e31fef5435bc634461fd2 - Cache (conancenter)
    meson/1.2.2#04bdfb85d665c82b08a3510aee3ffd19 - Cache (conancenter)
    ninja/1.11.1#77587f8c8318662ac8e5a7867eb4be21 - Cache (conancenter)
    pkgconf/2.1.0#27f44583701117b571307cf5b5fe5605 - Cache (conancenter)
Resolved version ranges
    zlib/[>=1.2.11 <2]: zlib/1.3.1

======== Computing necessary packages ========
gnu-config/cci.20210814: Current package revision is newer than the remote one
m4/1.4.19: Current package revision is newer than the remote one
zlib/1.3.1: WARN: Can't update, no package in remote
autoconf/2.71: Current package revision is newer than the remote one
automake/1.16.5: Current package revision is newer than the remote one
pkgconf/2.1.0: Current package revision is newer than the remote one
libtool/2.4.7: WARN: Can't update, no package in remote
libcurl/8.0.1: WARN: Can't update, no package in remote
cpr/1.10.0: Forced build from source
Requirements
    cpr/1.10.0#0ff0b00f5ea92fd3590d64c462477bd2:d9f9b96587f97a9f4a31080ed510f0b55ac9a84f - Build
    libcurl/8.0.1#6a17c16cc45f13b1ecbeca1f2dcd537e:267b943cf33985f9448f72c97e42fcdb5094fcf5#99fdb9dc3e2e6a70bcbc4e7db411f5ff - Cache
    zlib/1.3.1#f52e03ae3d251dec704634230cd806a2:2ee39e692ca4177b4b689b15bc1f2cfdf8f83706#f4afa4f31fadd974801b844e369437d3 - Cache
Build requirements
Skipped binaries
    autoconf/2.71, automake/1.16.5, gnu-config/cci.20210814, libtool/2.4.7, m4/1.4.19, meson/1.2.2, ninja/1.11.1, pkgconf/2.1.0

======== Installing packages ========
zlib/1.3.1: Already installed! (1 of 3)
libcurl/8.0.1: Already installed! (2 of 3)

-------- Installing package cpr/1.10.0 (3 of 3) --------
cpr/1.10.0: Building from source
cpr/1.10.0: Package cpr/1.10.0:d9f9b96587f97a9f4a31080ed510f0b55ac9a84f
cpr/1.10.0: Copying sources to build folder
cpr/1.10.0: Building your package in /Users/perseo/.conan2/p/b/cprb24099ada74f6/b
cpr/1.10.0: Calling generate()
cpr/1.10.0: Generators folder: /Users/perseo/.conan2/p/b/cprb24099ada74f6/b/build/Release/generators
cpr/1.10.0: CMakeToolchain generated: conan_toolchain.cmake
cpr/1.10.0: CMakeToolchain generated: /Users/perseo/.conan2/p/b/cprb24099ada74f6/b/build/Release/generators/CMakePresets.json
cpr/1.10.0: CMakeToolchain generated: /Users/perseo/.conan2/p/b/cprb24099ada74f6/b/src/CMakeUserPresets.json
cpr/1.10.0: CMakeDeps necessary find_package() and targets for your CMakeLists.txt
    find_package(CURL)
    target_link_libraries(... CURL::libcurl)
cpr/1.10.0: Generating aggregated env files
cpr/1.10.0: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
cpr/1.10.0: Calling build()
cpr/1.10.0: Apply patch (portability): disable warning flags and warning as error
cpr/1.10.0: Running CMake.configure()
cpr/1.10.0: RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/Users/perseo/.conan2/p/b/cprb24099ada74f6/p" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/perseo/.conan2/p/b/cprb24099ada74f6/b/src"
-- Using Conan toolchain: /Users/perseo/.conan2/p/b/cprb24099ada74f6/b/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)
-- Conan toolchain: C++ Standard 17 with extensions ON
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- The CXX compiler identification is AppleClang 15.0.0.15000309
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - broken
CMake Error at /opt/homebrew/Cellar/cmake/3.29.3/share/cmake/Modules/CMakeTestCXXCompiler.cmake:73 (message):
  The C++ compiler

    "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: '/Users/perseo/.conan2/p/b/cprb24099ada74f6/b/build/Release/CMakeFiles/CMakeScratch/TryCompile-2N1ABN'

    Run Build Command(s): /opt/homebrew/Cellar/cmake/3.29.3/bin/cmake -E env VERBOSE=1 /usr/bin/make -f Makefile cmTC_f4b74/fast
    /Applications/Xcode.app/Contents/Developer/usr/bin/make  -f CMakeFiles/cmTC_f4b74.dir/build.make CMakeFiles/cmTC_f4b74.dir/build
    Building CXX object CMakeFiles/cmTC_f4b74.dir/testCXXCompiler.cxx.o
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -D_GLIBCXX_USE_CXX11_ABI=0  -stdlib=libstdc++  -std=gnu++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk -fPIE -MD -MT CMakeFiles/cmTC_f4b74.dir/testCXXCompiler.cxx.o -MF CMakeFiles/cmTC_f4b74.dir/testCXXCompiler.cxx.o.d -o CMakeFiles/cmTC_f4b74.dir/testCXXCompiler.cxx.o -c /Users/perseo/.conan2/p/b/cprb24099ada74f6/b/build/Release/CMakeFiles/CMakeScratch/TryCompile-2N1ABN/testCXXCompiler.cxx
    clang: warning: include path for libstdc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]
    Linking CXX executable cmTC_f4b74
    /opt/homebrew/Cellar/cmake/3.29.3/bin/cmake -E cmake_link_script CMakeFiles/cmTC_f4b74.dir/link.txt --verbose=1
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -stdlib=libstdc++  -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names -L/usr/local/opt/llvm/lib  CMakeFiles/cmTC_f4b74.dir/testCXXCompiler.cxx.o -o cmTC_f4b74
    clang: warning: libstdc++ is deprecated; move to libc++ [-Wdeprecated]
    ld: warning: search path '/usr/local/opt/llvm/lib' not found
    ld: library 'stdc++' not found
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    make[1]: *** [cmTC_f4b74] Error 1
    make: *** [cmTC_f4b74/fast] Error 2





  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:2 (project)


-- Configuring incomplete, errors occurred!

cpr/1.10.0: ERROR:
Package 'd9f9b96587f97a9f4a31080ed510f0b55ac9a84f' build failed
cpr/1.10.0: WARN: Build folder /Users/perseo/.conan2/p/b/cprb24099ada74f6/b/build/Release
ERROR: cpr/1.10.0: Error in build() method, line 257
        cmake.configure()
        ConanException: Error 1 while executing
```

File: recipes/cpr/all/conanfile.py [hunk 1]
```diff
@@ -115,6 +115,9 @@ def validate(self):
         if Version(self.version) >= "1.9.0" and self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "6":
             raise ConanInvalidConfiguration(f"{self.ref} doesn't support gcc < 6")
 
+        if Version(self.version) >= "1.10.0" and self.settings.compiler in ("clang", "apple-clang") and self.settings.compiler.libcxx == "libstdc++":
```
Comment (RubenRBS) [inline recipes/cpr/all/conanfile.py:118]:
```suggestion
        if Version(self.version) >= "1.9.0" and self.settings.compiler in ("clang", "apple-clang") and self.settings.compiler.libcxx == "libstdc++":
```

File: recipes/cpr/all/conanfile.py [hunk 2]
```diff
@@ -115,6 +115,9 @@ def validate(self):
         if Version(self.version) >= "1.9.0" and self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "6":
             raise ConanInvalidConfiguration(f"{self.ref} doesn't support gcc < 6")
 
+        if Version(self.version) >= "1.9.0" and self.settings.compiler in ("clang", "apple-clang") and self.settings.compiler.libcxx == "libstdc++":
```
Comment (RubenRBS) [inline recipes/cpr/all/conanfile.py:118]:
```suggestion
        if Version(self.version) >= "1.6.2" and self.settings.compiler in ("clang", "apple-clang") and self.settings.compiler.libcxx == "libstdc++":
```

**Review by perseoGI — APPROVED** [review]
LGTM

**Pull request closed without merging**